autosubmit/job/job.py

```diff
@@ -2715,12 +2715,12 @@ def write_end_time(self, completed, count=-1):
                                                     job_id=self.id, out_file=out, err_file=err)
 
         # Launch second as threaded function only for slurm
-        if job_data_dc and type(self.platform) is not str and self.platform.type == "slurm":
-            thread_write_finish = Thread(target=ExperimentHistory(self.expid, jobdata_dir_path=BasicConfig.JOBDATA_DIR,
+        if job_data_dc and self.platform and self.platform.type == "slurm":
+            self.thread_write_finish = Thread(target=ExperimentHistory(self.expid, jobdata_dir_path=BasicConfig.JOBDATA_DIR,
```

Comment:
Huh, are we sure this is fixing it, @VindeeR ? I just checked and the thread is not a daemon thread here, so in theory I think the main process shouldn't exit until all non-daemon threads are gone?

https://docs.python.org/3/library/threading.html#:~:text=The%20significance%20of%20this%20flag%20is%20that%20the%20entire%20Python%20program%20exits%20when%20only%20daemon%20threads%20are%20left.